### PR TITLE
Added options for windowing

### PR DESCRIPTION
**Pull request by roxyboy**
It takes into account the issue raised in #16.

### CODE REVIEW

**Comment by rabernat** [comment]
Please rename this pull request to describe what it does. Does it fix a specific issue? If so, you can refer to the issue in the description (e.g. #16).

File: xrft/xrft.py [hunk 1]
```diff
@@ -13,10 +13,32 @@
...
+def _hanning(da, N, dim):
     """Apply Hanning window"""
 
-    window = np.hanning(N[-1]) * np.hanning(N[-2])[:, np.newaxis]
+    if len(N)==1:
+        window = xr.DataArray(np.hanning(N[0]), dims=dim)
+    elif len(N)==2:
+        window = xr.DataArray(np.hanning(N[-1])
+                            * np.hanning(N[-2])[:,np.newaxis],
+                            dims=dim)
```
Comment (rabernat) [inline xrft/xrft.py]: This if / else syntax can probably be avoided by using list comprehensions and reduce.

Comment (roxyboy) [inline xrft/xrft.py]: I'm not sure if that's possible as the broadcasting is different with data that have different dimensions.

**Comment by roxyboy** [comment]
I added tests for the _hanning function.

**Comment by rabernat** [comment]
#21 implements a different fix for the same issue.

**Comment by roxyboy** [comment]
Your new PR for windowing is better so I'm going to close this pull request.

**Pull request closed without merging**